fulltext/backends/__bin.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import absolute_import
+
+import string
+
+from six import StringIO
+
+try:
+    from string import maketrans
+
+    def bytes(chars):
+        return ''.join(map(chr, chars))
+
+except ImportError:
+    maketrans = bytes.maketrans
+
+
+BUFFER_MAX = 1024 * 1024
+
+DELETE = bytes([
+    i for i in range(256) if chr(i) not in string.printable
+])
+TRANSLATE = maketrans(b'\r\n', b'  ')
+
+
+def _get_file(f, **kwargs):
+    buffer = StringIO()
+
+    while True:
+        text = f.read(BUFFER_MAX)
+
+        if not text:
+            break
+
+        text = text.translate(TRANSLATE, DELETE)
+        text = text.decode('ascii', 'ignore')
+
+        # Emulate the `strings` CLI tool.
+        buffer.write(text)
+
+    return buffer.getvalue()
```

Comment:
Since you end up reading the whole file into memory anyway it looks like you don't need a separate `StringIO`, intermediate buffer and the while loop in general.

Reply:
I am going to refactor this library to return generators at some point in the near future. So text will be emitted in blocks. The `StringIO` buffers are temporary since the lib currently returns strings.

Reply:
OK, then LGTM

Reply:
Sorry, has historically returned strings. I also need to add the ability to return a separate string to indicate the "title" of a given file. Which can usually be obtained through metadata.